docs/core-concepts/database/sdk.mdx

```diff
@@ -1,270 +1,337 @@
-  .from('products')
-  .select()
-  .eq('category', 'electronics')
-  .or('on_sale.eq.true,rating.gte.4.5')
-  .order('price', { ascending: true })
+```typescript
+{
+  data: Array<object> | null,
+  error: Error | null,
+  count?: number
+}
 ```
 
-## Relationships & Joins
+<Warning>
+Always use filters like `.eq()` or `.in()` to specify which rows to update
+</Warning>
 
-The SDK supports PostgREST's powerful join syntax:
+### Examples
 
 <CodeGroup>
-```javascript Basic Join
-// Join with related data
+```javascript Update by ID
 const { data, error } = await insforge.database
-  .from('posts')
-  .select('*, users(*)')  // Include all user data
+  .from('users')
+  .update({ name: 'Jane Doe' })
+  .eq('id', userId)
+  .select()
 ```
 
-```javascript Specific Columns
-// Select specific columns from joined tables
+```javascript Update multiple
 const { data, error } = await insforge.database
-  .from('posts')
-  .select(`
-    id,
-    title,
-    content,
-    users (
-      nickname,
-      avatar_url
-    )
-  `)
+  .from('tasks')
+  .update({ status: 'completed' })
+  .in('id', ['task-1', 'task-2'])
+  .select()
 ```
+</CodeGroup>
 
-```javascript Aliased Joins
-// Aliased joins
-const { data, error } = await insforge.database
-  .from('posts')
-  .select('*, author:users(nickname, avatar_url)')
+### Output Example
+
+```json
+{
+  "data": [
+    { "id": "123", "name": "Jane Doe", "email": "john@example.com", "updated_at": "2024-01-15T11:00:00Z" }
+  ],
+  "error": null
+}
 ```
-</CodeGroup>
 
-## Counting & Pagination
+---
 
-Get total counts alongside your data:
+## delete()
 
-<CodeGroup>
-```javascript With Data
-// Get data with exact count
-const { data, count, error } = await insforge.database
-  .from('products')
-  .select('*', { count: 'exact' })
-  .eq('in_stock', true)
-  .range(0, 9)  // First 10 items
+Delete records from a table. Must use filters to target specific rows.
+
+### Parameters
+
+- `options.count` ('exact' | 'planned' | 'estimated', optional) - Include count of deleted rows
+
+### Returns
 
-console.log(`Showing ${data.length} of ${count} products`)
+```typescript
+{
+  data: Array<object> | null,
+  error: Error | null,
+  count?: number
+}
 ```
 
-```javascript Count Only
-// Count only (no data)
-const { count, error } = await insforge.database
-  .from('products')
-  .select('*', { count: 'exact', head: true })
-  .eq('category', 'electronics')
+<Warning>
+Always use filters to specify which rows to delete
+</Warning>
+
+### Examples
+
+<CodeGroup>
+```javascript Delete by ID
+const { error } = await insforge.database
+  .from('posts')
+  .delete()
+  .eq('id', postId)
+```
+
+```javascript Delete with filter
+const { error } = await insforge.database
+  .from('sessions')
+  .delete()
+  .lt('expires_at', new Date())
 ```
 </CodeGroup>
 
-## Error Handling
+### Output Example
+
+```json
+{
+  "data": null,
+  "error": null
+}
+```
+
+---
 
-The SDK provides structured error responses:
+## Filters
```

Comment:
maybe introducing API under the order of insert - update - delete - query makes more sense, thus select and filter statements are more closely connected.